src/main/java/io/appform/dropwizard/sharding/metrics/TransactionMetricManager.java

```diff
@@ -0,0 +1,88 @@
+package io.appform.dropwizard.sharding.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SlidingTimeWindowArrayReservoir;
+import com.codahale.metrics.Timer;
+import com.google.common.base.Strings;
+import io.appform.dropwizard.sharding.config.MetricConfig;
+import io.appform.dropwizard.sharding.execution.TransactionExecutionContext;
+import lombok.val;
+
+import java.util.concurrent.TimeUnit;
+
+public class TransactionMetricManager {
+
+    private MetricConfig metricConfig;
+    private MetricRegistry metricRegistry;
+    private static final String METRIC_PREFIX = "db.sharding";
+    private static final String DELIMITER = ".";
+    private static final String DELIMITER_REPLACEMENT = "_";
+
+    public void initialize(final MetricConfig metricConfig,
```

Comment:
Is this config dynamic ?

Reply:
Done. Added a method in dbShardingBase to take metric config supplier with default impl of static config